crates/storage/trie/trie.rs

```diff
@@ -154,59 +168,56 @@ impl Trie {
         Ok(node_path)
     }
 
-    pub fn verify_proof(
-        _proof: &[Vec<u8>],
-        _root_hash: NodeHash,
-        _path: &PathRLP,
-        _value: &ValueRLP,
-    ) -> Result<bool, TrieError> {
-        // This is a mockup function for verifying proof of inclusions. This function will be
-        // possible to implement after refactoring the current Trie implementation.
+    /// Obtains all encoded nodes traversed until reaching the node where every path is stored.
+    /// The list doesn't include the root node, this is returned separately.
+    /// Will still be constructed even if some path is not stored in the trie.
+    pub fn get_pruned_state(
```

Comment:
Can't we re-use `get_proof` for this method? the logic seems quite similar. Also the name is not very descriptive, maybe we could name it `get_proofs` instead?

Reply:
I agree! done.